swig/include/python/docs/ogr_layer_docs.i

```diff
@@ -1,1770 +1,759 @@
+For more details: :cpp:func:`OGR_L_SetStyleTableDirectly`
+";
 
-Set style table. ";
+%feature("docstring")  SetStyleTable "
+Set style table.
 
-%feature("docstring")  GetName "const char\\* OGR_L_GetName(OGRLayerH
-hLayer)
+For more details: :cpp:func:`OGR_L_SetStyleTable`
+";
 
+%feature("docstring")  GetName "
 Return the layer name.
 
-This returns the same content as
-OGR_FD_GetName(OGR_L_GetLayerDefn(hLayer)), but for a few drivers,
-calling OGR_L_GetName() directly can avoid lengthy layer definition
-initialization.
-
-This function is the same as the C++ method OGRLayer::GetName().
+For more details: :cpp:func:`OGR_L_GetName`
 
 .. versionadded:: 1.8.0
 
-Parameters
------------
-hLayer:
-    handle to the layer.
-
 Returns
 --------
 str:
-    the layer name (must not been freed)
+    The layer name
 ";
 
-%feature("docstring")  GetGeomType "OGRwkbGeometryType
-OGR_L_GetGeomType(OGRLayerH hLayer)
-
+%feature("docstring")  GetGeomType "
 Return the layer geometry type.
 
-This returns the same result as
-OGR_FD_GetGeomType(OGR_L_GetLayerDefn(hLayer)), but for a few drivers,
-calling OGR_L_GetGeomType() directly can avoid lengthy layer
-definition initialization.
-
-For layers with multiple geometry fields, this method only returns the
-geometry type of the first geometry column. For other columns, use
-OGR_GFld_GetType(OGR_FD_GetGeomFieldDefn(OGR_L_GetLayerDefn(hLayer),
-i)). For layers without any geometry field, this method returns
-wkbNone.
-
-This function is the same as the C++ method OGRLayer::GetGeomType().
+For more details: :cpp:func:`OGR_L_GetGeomType`
 
 .. versionadded:: 1.8.0
 
-Parameters
------------
-hLayer:
-    handle to the layer.
-
 Returns
 --------
-OGRwkbGeometryType:
-    the geometry type
+int:
+    The geometry type code. The types can be found with
+    'osgeo.ogr.wkb' prefix. For example :py:const:`osgeo.ogr.wkbPolygon`.
 ";
 
-%feature("docstring")  SetIgnoredFields "OGRErr
-OGR_L_SetIgnoredFields(OGRLayerH hLayer, const char \\*\\*papszFields)
-
+%feature("docstring")  SetIgnoredFields "
 Set which fields can be omitted when retrieving features from the
 layer.
 
-If the driver supports this functionality (testable using
-OLCIgnoreFields capability), it will not fetch the specified fields in
-subsequent calls to GetFeature() / GetNextFeature() and thus save some
-processing time and/or bandwidth.
-
-Besides field names of the layers, the following special fields can be
-passed: \'OGR_GEOMETRY\' to ignore geometry and \'OGR_STYLE\' to
-ignore layer style.
-
-By default, no fields are ignored.
-
-This method is the same as the C++ method OGRLayer::SetIgnoredFields()
+For more details: :cpp:func:`OGR_L_SetIgnoredFields`
 
 Parameters
 -----------
-hLayer:
-    handle to the layer
-papszFields:
-    an array of field names terminated by NULL item.
-    If NULL is passed, the ignored list is cleared.
+options: list[str]
+    A list of field names.
+    If None is passed, the ignored list is cleared.
```

Comment:
that's true for C/C++ but not Python. The current typemap doesn't support None. Empty list [] should rather be used